client/src/modules/Admin/Components/ManageAdminModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react'
+import axios from 'axios'
+import styles from '../Styles/ManageAdminModal.module.css'
+
+import { Student } from 'common'
+import AdminUser from './AdminUser'
+import closeIcon from '../../../assets/icons/X.svg'
+
+type Props = {
+    open: boolean
+    setOpen: (open: boolean) => void
+    token: string
+  }
+
+const ManageAdminModal = ({token, open, setOpen}: Props) => {
+    const [admins, setAdmins] = useState<Student[]>([])
+    const [netId, setNetId] = useState<string>("")
+
+    function closeModal() {
+        setOpen(false)
+    }
+
+    /**
+     * Endpoint to get all admins
+     */
+    useEffect(() => {
+        axios
+        .post('/api/getAdmins', {token: token})
+        .then((response) => {
+          const result = response.data.result
+          if (response.status === 200) {
+            setAdmins(result)
+          } else {
+            console.log('Error at getAdmins')
+          }
+        })
+      }, [token])
+
+    /**
+     * Removes an admin from the list, giving that user 'regular' privilege
+     * @param user assumes that this user already has admin privilege
+     */
+
+    function removeAdmin(user: Student) {
+        axios
+        .post('/api/removeAdmin', {
+            userId: user._id,
+            token: token
+        })
+        .then((response) => {
+            if (response.status === 200) {
+                const updatedAdmins = admins.filter((admin: Student) => {
+                    return admin && admin._id !== user._id
+                })
+                setAdmins(updatedAdmins)
+            }
+        }).catch((e) => console.log(`Unable to remove admin ${e}`))
+    }
+
+    /**
+     * Calls endpoint to add or update a user with admin privilege
+     * @param _netId the user's net id
+     */
+
+    function addAdminByNetId(_netId: string) {
```

Comment:
Great work with this new feature! It'll help us better manage admin access semester-to-semester. I've left a couple comments with things I noticed while testing.

Currently, adding the admin by netid doesn't automatically update the frontend - perhaps use `useEffect()`? I think the remove function also doesn't automatically update.

Also, it'd be nice to attach the admin's full name before the netid (should just be a quick query to the name in the database)

I'm not sure if this is feasible, but restricting who can manage the admins would be great (maybe just permissions for TPM). Along with this, making sure that the admin manager shouldn't be able to remove themselves or add duplicates of other members on the team will help prevent "lockouts" and access issues.